### ISSUE TITLE
DO unable to change hostname correctly in Google Cloud

### ISSUE THREAD

**Issue by nitinthewiz** [F5Networks/f5-declarative-onboarding#292]
### Environment
 * Declarative Onboarding Version: 1.27.0
 * BIG-IP Version: 16.1.0-0.0.19

### Summary
While running DO as part of runtime-init (as part of the startup script) on our templates in Google Cloud, we observe a consistent failure where DO fails to "find the requested system device" by it's hostname. Our DO config is supposed to change the hostname from xyz.internal to xyz.local. However, we observe that it fails to do so and looks for the device by the name xyz.internal. When it fails to find it, DO fails and runtime-init fails as well. 

However, when we run the runtime-init command manually afterwards, it completes without any failures and without any complaints from DO. Adding a sleep of 10-30 seconds also doesn't seem to help DO complete it's run in the first shot as part of the startup-script.

### Steps To Reproduce
1. Submit the following declaration:
```json
"declaration": {
        "schemaVersion": "1.0.0",
        "class": "Device",
        "label": "Autoscale 1NIC BIG-IP declaration for Declarative Onboarding with BIG-IQ license",
        "async": true,
        "Common": {
            "class": "Tenant",
            "My_DbVariables": {
                "class": "DbVariables",
                "provision.extramb": 1000,
                "restjavad.useextramb": true,
                "ui.advisory.color": "blue",
                "ui.advisory.text": "Provisioned via Runtime Init and DO"
            },
            "My_Dns": {
                "class": "DNS",
                "nameServers": [
                    "x.x.x.x"
                ]
            },
            "My_License": {
                "class": "License",
                "hypervisor": "gce",
                "licenseType": "licensePool",
                "licensePool": "production",
                "bigIqHost": "x.x.x.x",
                "bigIqUsername": "admin",
                "tenant": "tenant_name",
                "skuKeyword1": "LICENSE_SKU",
                "unitOfMeasure": "hourly",
                "reachable": false,
                "overwrite": false
            },
            "My_Ntp": {
                "class": "NTP",
                "servers": [
                    "0.pool.ntp.org"
                ],
                "timezone": "UTC"
            },
            "My_Provisioning": {
                "asm": "nominal",
                "class": "Provision",
                "ltm": "nominal"
            },
            "My_System": {
                "class": "System",
                "autoPhonehome": true,
                "hostname": "dd-hostname.local",
                "consoleInactivityTimeout": 0,
                "cliInactivityTimeout": 0,
                "autoCheck": true,
                "tmshAuditLog": true,
                "guiAuditLog": false,
                "mcpAuditLog": "enable",
                "preserveOrigDhcpRoutes": false
            },
            "admin": {
                "class": "User",
                "shell": "bash",
                "userType": "regular"
            }
        }
    }
```

2. Observe the following error response:
```json
{
    "id": "c0306f89-a0d6-439e-8c19-6abd2bdc1063",
    "selfLink": "https://localhost/mgmt/shared/declarative-onboarding/task/c0306f89-a0d6-439e-8c19-6abd2bdc1063",
    "code": 404,
    "status": "ERROR",
    "message": "invalid config - rolled back",
    "errors": [
        "tryUntil: max tries reached: 01020036:3: The requested system device (/Common/dd-hostname.c.orgname.internal) was not found.",
        "tryUntil: max tries reached: 01020036:3: The requested system device (/Common/dd-hostname.c.orgname.internal) was not found."
    ],
    "result": {
        "class": "Result",
        "code": 404,
        "status": "ERROR",
        "message": "invalid config - rolled back",
        "errors": [
            "tryUntil: max tries reached: 01020036:3: The requested system device (/Common/dd-hostname.c.orgname.internal) was not found.",
            "tryUntil: max tries reached: 01020036:3: The requested system device (/Common/dd-hostname.c.orgname.internal) was not found."
        ]
    },
    "declaration": {}
}
```

### Expected Behavior
DO should be able to change the hostname in the first shot and not fail.

### Actual Behavior
DO fails with the error that it could not find the system device with the old hostname. It seems to be successful in changing the hostname, but when it lists the info in restnoded (as shown below) it doesn't seem to have changed the name.

Please include any applicable error output.

### LOG - restnoded.log
```
Thu, 10 Feb 2022 19:28:28 GMT - fine: [f5-declarative-onboarding: restWorker.js] modify localhost /tm/sys/global-settings {"hostname":"dd-hostname.local"}
Thu, 10 Feb 2022 19:28:28 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"kind":"tm:sys:global-settings:global-settingsstate","selfLink":"https://localhost/mgmt/tm/sys/global-settings?ver=16.1.0","awsApiMaxConcurrency":1,"consoleInactivityTimeout":0,"customAddr":"none","failsafeAction":"go-offline-restart-tm","fileBlacklistPathPrefix":"{/shared/3dns/} {/shared/bin/} {/shared/core/} {/appdata/core/} {/shared/datasync/} {/shared/em/} {/shared/GeoIP/} {/shared/images/} {/shared/lib/} {/shared/lib64/} {/shared/log/} {/shared/lost+found/} {/shared/mgmt/} {/shared/nfb/} {/shared/ssh/} {/shared/statsd/} {/shared/tmstat/} {/shared/vadc/} {/config/aaa/} {/config/big3d/} {/config/bigip/} {/config/filestore/} {/config/gtm/} {/config/httpd/} {/config/ntp.conf} {/config/rndc.key} {/config/ssh/} {/config/ssl/}","fileBlacklistReadOnlyPathPrefix":"{/etc/shadow}","fileLocalPathPrefix":"{/shared/} {/tmp/}","fileWhitelistPathPrefix":"{/var/local/scf} {/tmp/} {/shared/} {/config/} {/usr/share/aws/}","guiAudit":"disabled","guiExpiredCertAlert":"enabled","guiSecurityBanner":"enabled","guiSecurityBannerText":"Welcome to the BIG-IP Configuration Utility.\n\nLog in with your username and password using the fields on the left.","guiSetup":"disabled","hostAddrMode":"management","hostname":"dd-hostname.local","lcdDisplay":"enabled","ledLocator":"disabled","mgmtDhcp":"enabled","netReboot":"disabled","passwordPrompt":"********","quietBoot":"enabled","sshMaxSessionLimit":10,"sshMaxSessionLimitPerUser":-1,"sshRootSessionLimit":"disabled","sshSessionLimit":"disabled","usernamePrompt":"Username"}
Thu, 10 Feb 2022 19:28:28 GMT - fine: [f5-declarative-onboarding: systemHandler.js] Checking Users.
Thu, 10 Feb 2022 19:28:30 GMT - finest: [f5-declarative-onboarding: restWorker.js] /tm/auth/user/admin exists, modifying
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] modify localhost /tm/auth/user/admin {"name":"admin","password":"********","shell":"bash"}
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"kind":"tm:auth:user:userstate","name":"admin","fullPath":"admin","generation":0,"selfLink":"https://localhost/mgmt/tm/auth/user/admin?ver=16.1.0","description":"Admin User","encryptedPassword":"********","sessionLimit":-1,"shell":"bash","partitionAccess":[{"name":"all-partitions","role":"admin","nameReference":{"link":"https://localhost/mgmt/tm/auth/partition/all-partitions?ver=16.1.0"}}]}
Thu, 10 Feb 2022 19:28:30 GMT - info: [f5-declarative-onboarding: restWorker.js] This is a BIG-IP
Thu, 10 Feb 2022 19:28:30 GMT - info: [f5-declarative-onboarding: restWorker.js] Waiting for device to be ready.
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/echo/available
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/identified-devices/config/device-info/available
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/sys/available
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/cm/available
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/iapp/package-management-tasks/available
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/sys/mcp-state
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"kind":"tm:sys:mcp-state:mcp-statestats","selfLink":"https://localhost/mgmt/tm/sys/mcp-state?ver=16.1.0","entries":{"https://localhost/mgmt/tm/sys/mcp-state/0":{"nestedStats":{"entries":{"endPlatformIdReceived":{"description":"true"},"lastLoad":{"description":"full-config-load-succeed"},"phase":{"description":"running"}}}}}}
Thu, 10 Feb 2022 19:28:30 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/identified-devices/config/device-info
Thu, 10 Feb 2022 19:28:30 GMT - finest: socket 428 opened
Thu, 10 Feb 2022 19:28:31 GMT - fine: [f5-declarative-onboarding: restWorker.js] {"baseMac":"{}","hostMac":"{}","halUuid":"{}","chassisSerialNumber":"{}","slots":[{"volume":"HD1.1","product":"BIG-IP","version":"16.1.0","build":"0.0.19","isActive":true}],"license":{"generation":0,"lastUpdateMicros":1644521209391999},"interfaces":["1.0","mgmt"],"isIControlRestSupported":true,"icrdPort":8100,"time":1644521311307,"physicalMemory":30720,"platform":"Z100","cpu":"Intel(R) Xeon(R) CPU @ 2.00GHz","machineId":"{}","address":"{}","hostname":"dd-hostname.c.orgname.internal","version":"16.1.0","product":"BIG-IP","platformMarketingName":"BIG-IP Virtual Edition","edition":"Final","build":"0.0.19","restFrameworkVersion":"16.1.0-0.0.19","managementAddress":"{}","mcpDeviceName":"/Common/bigip1","isClustered":false,"isVirtual":true,"hypervisorType":"gce","generation":0,"lastUpdateMicros":0,"kind":"shared:resolver:device-groups:deviceinfostate","selfLink":"https://localhost/mgmt/shared/identified-devices/config/device-info"}
```

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added this to our internal product backlog as AUTOTOOL-3012.

**Comment by dstokesf5** [comment]
This issue has been resolved with the 1.30.0 release.
